django/tasks/task.py

```diff
@@ -0,0 +1,271 @@
+from dataclasses import dataclass, field, replace
+from datetime import datetime
+from inspect import iscoroutinefunction
+from typing import Any, Callable, Dict, Optional, Type
+
+from asgiref.sync import async_to_sync, sync_to_async
+
+from django.db.models.enums import TextChoices
+from django.utils.translation import gettext_lazy as _
+
+from .exceptions import ResultDoesNotExist
+from .utils import get_module_path, json_normalize
+
+DEFAULT_TASK_BACKEND_ALIAS = "default"
+DEFAULT_QUEUE_NAME = "default"
+MIN_PRIORITY = -100
+MAX_PRIORITY = 100
+DEFAULT_PRIORITY = 0
+
+TASK_REFRESH_ATTRS = {
+    "_exception_class",
+    "_traceback",
+    "_return_value",
+    "finished_at",
+    "started_at",
+    "status",
+    "enqueued_at",
+}
+
+
+class ResultStatus(TextChoices):
+    NEW = ("NEW", _("New"))
+    RUNNING = ("RUNNING", _("Running"))
+    FAILED = ("FAILED", _("Failed"))
+    SUCCEEDED = ("SUCCEEDED", _("Succeeded"))
+
+
+@dataclass(frozen=True)
```

Comment:
Also consider `slots=True`, `kw_only=True` (though need to check the Python version they were added...)

Reply:
I'd originally intentionally left them out to keep support for 3.9, but as that ends in 4 months, and Django 5.2 already dropped official support for it, now seems like the perfect time to add them!